homeassistant/components/solaredge/coordinator.py

```diff
@@ -90,12 +90,17 @@ def update(self) -> None:
         # Sanity check the energy values. SolarEdge API sometimes report "lifetimedata" of zero,
         # while values for last Year, Month and Day energy are still OK.
         # See https://github.com/home-assistant/core/issues/59285 .
+        # Use a tolerance factor so that we don't fail by small variances.
+        # See https://github.com/home-assistant/core/issues/69600 .
+        validity_tolerance = 0.9
         if set(energy_keys).issubset(self.data.keys()):
             for index, key in enumerate(energy_keys, start=1):
                 # All coming values in list should be larger than the current value.
-                if any(self.data[k] > self.data[key] for k in energy_keys[index:]):
-                    self.data = {}
-                    raise UpdateFailed("Invalid energy values, skipping update")
+                if any(
+                    self.data[k] * validity_tolerance > self.data[key]
+                    for k in energy_keys[index:]
+                ):
+                    self.data.pop(key)
```

Comment:
Considering all this logic, shouldn't we just filter out `0` values?

I feel like adding things like tolerance is not how things should work.

Reply:
The reason why I didn't do that is because some user reported problem was not only related to zero-values. For instance:
https://github.com/home-assistant/core/issues/59285#issuecomment-1071614709

However, I do agree that adding a tolerance to a validation check is not an elegant solution. The only other solution I can thing of is comparing with previous value and only update if larger. But that will likely open up for weird corner cases. Also that kind of filtering should maybe be done on a higher level in the energy dashboard code instead.

Reply:
> Also that kind of filtering should maybe be done on a higher level in the energy dashboard code instead.

Actually, we had tolerance and such logic in the beginning and removed it.

Reply:
Is there any chance this fix will make it into 2022.5.5? HA is amazing but with the SolarEdge integration not working in recent versions I'm feeling a bit stuck. Thank you for all your work on this.

Reply:
I've removed tolerance factor now. Now we will just ignore the specific entities that has invalid values, keeping the ones that  are OK. And also without printing the ugly error message. Hopefully good enough for now.